src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -345,7 +345,25 @@ private HttpMessageHandler SetupHandlerChain()
                 return Task.FromException<HttpResponseMessage>(error);
             }
 
-            return handler.SendAsync(request, cancellationToken);
+            if (HttpTelemetry.IsEnabled && request.RequestUri != null)
+            {
+                // Wrap the task for event activity-id threading.
+                HttpTelemetry.Log.RequestStart(request.RequestUri.IdnHost, request.RequestUri.Port);
+                Task<HttpResponseMessage> task = handler.SendAsync(request, cancellationToken);
```

Comment:
`SendAsync` might throw synchronously and then no `RequestStop` would be logged.
For example when the `IWebProxy` implementation throws or for example here: https://github.com/dotnet/runtime/blob/2b7852b3d039b7d40f2623a3ec54e7d23eb93790/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs#L379